errbot/core_plugins/plugins.py

```diff
@@ -17,24 +19,27 @@ def repos_install(self, mess, args):
         or an url towards a tar.gz archive : http://www.gootz.net/plugin-latest.tar.gz
         """
         if not args.strip():
-            return "You should have an urls/git repo argument"
-        self._bot.repo_manager.install_repo(args)
-
-        errors = self._bot.plugin_manager.update_dynamic_plugins()
-        if errors:
-            self.send(mess.frm, 'Some plugins are generating errors:\n' + '\n'.join(errors),
-                      message_type=mess.type)
-        else:
-            self.send(
-                mess.frm,
-                ("A new plugin repository has been installed correctly from "
-                 "%s. Refreshing the plugins commands..." % args),
-                message_type=mess.type
-            )
-        loading_errors = self._bot.plugin_manager.activate_non_started_plugins()
-        if loading_errors:
-            return loading_errors
-        return "Plugins reloaded without any error."
+            yield "You should have an urls/git repo argument"
+            return
+        try:
+            local_path = self._bot.repo_manager.install_repo(args)
+            errors = self._bot.plugin_manager.update_dynamic_plugins()
+            if errors:
+                yield 'Some plugins are generating errors:\n' + '\n'.join(errors.values())
+                # if the load of the plugin failed, uninstall cleanly teh repo
+                for path in errors.keys():
+                    if path.startswith(local_path):
+                        yield 'Removing %s as it did not load correctly.' % local_path
+                        shutil.rmtree(local_path)
```

Comment:
So this is what I am doing if there is no bug: 
local_path = self._bot.repo_manager.install_repo(args)   # gets the path to the repo this command is installing

if path.startswith(local_path)  # only remove the one we are talking about.